src/ConnectionTransports/NetworkSocketConnectionTransport.cpp

```diff
@@ -25,335 +25,222 @@ NetworkSocketConnectionTransport::NetworkSocketConnectionTransport(
     connectionKind(kind),
     socketHandle(socketHandle),
     targetAddr(targetAddr)
-{ }
-#pragma endregion Constructor/Destructor
-
-#pragma region ConnectionTransport Implementation
-std::optional<sockaddr_in> NetworkSocketConnectionTransport::GetAddr()
-{
-    return targetAddr;
-}
-
-std::optional<sockaddr_in6> NetworkSocketConnectionTransport::GetAddr6()
-{
-    // TODO
-    return std::nullopt;
-}
-
-Result<void> NetworkSocketConnectionTransport::StartAsync()
 {
+    
     // First, set the socket to non-blocking IO mode
     int socketFlags = fcntl(socketHandle, F_GETFL, 0);
     if (socketFlags == -1)
     {
         int error = errno;
         Stop();
-        return Result<void>::Error(
-            fmt::format(
-                "Could not retrieve socket flags. Error {}: {}",
-                error,
-                Util::ErrnoToString(error)));
+        // TODO throw
```

Comment:
Just leaving a comment here to track this

Reply:
Addressed.